R/mzQCjson.R

```diff
@@ -0,0 +1,255 @@
+library(stringr)
+library(readr)
+
+#'
+#' Get the information of each cv Term from a obo file.
+#' 
+#' 
+#' 
+#' @param file "xxx.obo"
+#' @return A list containing cv Term's information
+#' 
+#' @export
+#' 
+#' 
+#' 
+read_cvTerm <- function(file){
+  obo <- data.frame(read.csv(file = file, header = FALSE))
+  cv_Term <- list(cvRef = list(),accession= list(), name = list(), units = list())
+  term_index <- 0
+  for (x in obo$V1) {
+    if(x == "[Term]"){
+      term_index = term_index + 1
+      cv_Term$units[term_index] = ""
+    }
+    if(substring(x,1,2)  == "id"){
+      cv_Term$cvRef[term_index] = substring(str_sub(x, str_locate(x, ":")[1]+2), 1,2)
+      cv_Term$accession[term_index] = str_sub(x, str_locate(x, ":")[1]+2)
+    }
+    if(substring(x,1,4)  == "name"){
+      cv_Term$name[term_index] = str_sub(x, str_locate(x, ":")[1]+2)
+
+    }
+    if(substring(x,1,4)  == "is_a"){
+      ## Connect multiple unit through commas
+      cv_Term$units[term_index] = paste(cv_Term$units[term_index], str_sub(x, str_locate(x, ":")[1]+2), seq=",")
+
+    }
+  }
+  ##Divide the units joined by commas into a list
+  cv_Term$units <- lapply(cv_Term$units, function(x){
+    if(x != ""){
+      
+      units <- unlist(strsplit(x , split = ","))
+      data <- (lapply(units, function(y){
+        cvRef <- str_sub(y, 1, str_locate(y, "\\:")[1]-1)
+        accession <- str_sub(y, 1, str_locate(y,"\\ !")[1]-1)
+        name <- str_sub(y, str_locate(y, "\\! ")[1]+2)
+        return(list(cvRef, accession, name))
+      }))
+      return(rjson::toJSON(data))
+      
+    }else{
+      data <- NA
+      return(data)
+    }
+  })
+  return(cv_Term)
+}
+
+
+#'
+#' Get all information needed in metric (cvRef, accession, name, value, unit, type of quality, input_file),
+#' type of quality and input_file act on the connection to the other two tables(inputfile_info, analysisSoftware)
+#' that belong to metadata
+#' 
+#' 
+#' 
+#' @param qcMetric A list containing all metric information
+#' @param data data frame of all cv Parameter
```

Comment:
can you think of a better (more descriptive) name for `data`?

Reply:
how about `qc_cv` instead of `data`?

Reply:
yes, great!
Maybe add the column names that it should have as well?

Reply:
add the column names ?

Reply:
yes. add the names that you expect the data.frame to have to the documentation

Reply:
or better, check them inside the function before doing any computation on them. 
Similar to what the qcMetrics are doing on their input